### PR TITLE
[5.2] Refactor all instances of File::delete() to use framework

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
We are in the process of migrating away from the CMS filesystem package to use the framework version of it. This PR converts all instances where File::delete() is used to use the framework version. In that process, all other calls to the CMS File class in that file were converted to properly handle possible exceptions from the framework equivalent. This intentionally did not add additional error handling. I expect that to be a separate effort in the future.

### Testing Instructions
Codereview.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Console/UpdateCoreCommand.php [hunk 1]
```diff
@@ -280,13 +281,20 @@ private function updateJoomlaCore($updatemodel): bool
...
-                if (file_exists($autoloadFile)) {
-                    File::delete($autoloadFile);
-                }
-
-                // Remove the xml
-                if (file_exists(JPATH_BASE . '/joomla.xml')) {
-                    File::delete(JPATH_BASE . '/joomla.xml');
+                try {
+                    if (file_exists($autoloadFile)) {
+                            File::delete($autoloadFile);
```
Comment (Quy) [inline libraries/src/Console/UpdateCoreCommand.php]:
```suggestion
                         File::delete($autoloadFile);
```

File: libraries/src/Console/UpdateCoreCommand.php [hunk 2]
```diff
@@ -280,13 +281,20 @@ private function updateJoomlaCore($updatemodel): bool
...
-                if (file_exists($autoloadFile)) {
-                    File::delete($autoloadFile);
-                }
-
-                // Remove the xml
-                if (file_exists(JPATH_BASE . '/joomla.xml')) {
-                    File::delete(JPATH_BASE . '/joomla.xml');
+                try {
+                    if (file_exists($autoloadFile)) {
+                         File::delete($autoloadFile);
```
Comment (Quy) [inline libraries/src/Console/UpdateCoreCommand.php]:
```suggestion
                        File::delete($autoloadFile);
```

File: plugins/task/rotatelogs/src/Extension/RotateLogs.php [hunk 3]
```diff
@@ -100,7 +101,11 @@ private function rotateLogs(ExecuteTaskEvent $event): int
             if ($version >= $logsToKeep) {
                 // Delete files which have version greater than or equals $logsToKeep
                 foreach ($files as $file) {
-                    File::delete($logPath . '/' . $file);
+                    try {
+                        File::delete($logPath . '/' . $file);
+                    } catch (FilesystemException $exception) {
+
+                    }
```
Comment (HLeithner) [inline plugins/task/rotatelogs/src/Extension/RotateLogs.php]: Feels wrong to me, does the task scheduler has no logging?

Comment (Hackwar) [reply plugins/task/rotatelogs/src/Extension/RotateLogs.php]: This is a direct conversion of the current code. Additional changes are outside of the scope of this PR. Let's do additional improvements in another PR.

**Pull request merged**